File: an2_sem4/MPP/Homework/sapt78/java/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/dto/BiletDTO.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.dto;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Entity;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;

public class BiletDTO extends EntityDTO {
    private final String numeCumparator;
    private final int nrLocuri;
    private final SpectacolDTO spectacol;

    public BiletDTO(String numeCumparator, int nrLocuri, SpectacolDTO spectacol) {
        this.numeCumparator = numeCumparator;
        this.nrLocuri = nrLocuri;
        this.spectacol = spectacol;
    }

    public String getNumeCumparator() {
        return numeCumparator;
    }

    public int getNrLocuri() {
        return nrLocuri;
    }

    public SpectacolDTO getSpectacol() {
        return spectacol;
    }
}

File: an2_sem4/MPP/Homework/sapt78/cs/FestivalSellpoint/FestivalSellpoint.Network/DTO/BiletDTO.cs
using FestivalSellpoint.Domain;
using System;

namespace FestivalSellpoint.Network.DTO
{
    [Serializable]
    public class BiletDTO
    {
        public string NumeCumparator { get; }
        public int NrLocuri { get; }
        public SpectacolDTO Spectacol { get; }

        public BiletDTO(string numeCumparator, int nrLocuri, Spectacol spectacol)
        {
            NumeCumparator = numeCumparator;
            NrLocuri = nrLocuri;
            Spectacol = SpectacolDTO.FromSpectacol(spectacol);
        }
    }
}
